src/util/ieee_float.cpp

```diff
@@ -44,32 +43,32 @@ float u32_FloatSlow(uint32_t i)
 
 	if (!exp) {
 		// Denormal or zero
-		return sign ? -ldexpf((float)imant, -149) : ldexpf((float)imant, -149);
+		return sign ? -ldexpf((f32)imant, -149) : ldexpf((f32)imant, -149);
 	}
 
-	return sign ? -ldexpf((float)(imant | 0x800000), exp - 150) :
-		ldexpf((float)(imant | 0x800000), exp - 150);
+	return sign ? -ldexpf((f32)(imant | 0x800000), exp - 150) :
+		ldexpf((f32)(imant | 0x800000), exp - 150);
 }
 
-// Given a float, return an unsigned 32-bit integer representing the float
+// Given a float, return an unsigned 32-bit integer representing the f32
 // in IEEE-754 single-precision format.
-uint32_t float_u32Slow(float f)
+u32 f32Tou32Slow(f32 f)
 {
-	uint32_t signbit = copysignf(1.0f, f) == 1.0f ? 0 : 0x80000000UL;
-	uint32_t imant;
+	u32 signbit = copysignf(1.0f, f) == 1.0f ? 0 : 0x80000000UL;
+	u32 imant;
 	int exp;
 	if (f == 0.f)
 		return signbit;
 	if (isnanf(f))
 		return signbit | 0x7FC00000UL;
 	if (isinff(f))
 		return signbit | 0x7F800000UL;
-	float mant = frexpf(f, &exp);
-	imant = (uint32_t)floorf((signbit ? -16777216.f : 16777216.f) * mant);
+	f32 mant = frexpf(f, &exp);
+	imant = (u32)floorf((signbit ? -16777216.f : 16777216.f) * mant);
 	exp += 126;
 	if (exp <= 0) {
 		// Denormal
-		return signbit | (imant >> 1 - exp);
+		return signbit | (imant >> (1 - exp));
```

Comment:
This shift might overflow (invoking Undefined Behaviour) if the exponent is lower than that supported by IEEE. I think that could be solved as follows:
```C++
		return signbit | (exp <= -31 ? 0 : imant >> (1 - exp));
```
Denormals should be rare, therefore this should not be an important performance drag.